### DISCUSSION TITLE
How do I retrieve media by uuid?

### DISCUSSION THREAD

**Discussion by michaeljcoyne** [spatie/laravel-medialibrary#2411] [Q&A]
I have tried
$modelClass->getMedia('handouts', ['uuid' => 'myuuid']);
To get specific file using the filter. Doesn't work.
What is the correct way? 
I cactially want to retrieve and delete this file.
Thanks
Michael

**Comment by michaeljcoyne** [comment]
I have actually done it with   $modelClass->deleteMedia($id); 
But still interested to know if ti can be acieved other way.
Thanks
Michael

**Comment by aricooperdavis** [comment]
I don't know if there is a built-in way to do this, but you could always loop over the media associated with your model:

```php
$targetUuid = '0638a4ae-8fc0-4b98-b95d-01689d1f95a4';
$modelMedia = $model->getMedia('collection');
foreach($modelMedia as $key => $value) {
  if ($value->uuid == $targetUuid) {
    $value->delete();
    break;
  }
}
```
